// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////
syntax = "proto3";

package com.google.cloud.sme.common;

option java_package = "com.google.cloud.sme";
option java_outer_classname = "Entities";

import "google/protobuf/timestamp.proto";

message User {
  int64 id = 1;
  string first_name = 2;
  string last_name = 3;
  string country = 4;
}

message Item {
  int64 id = 1;
  string size = 2;
  string color = 3;
  string type = 4;
  double price = 5;
}

message Action {
  enum ActionType {
    UNKNOWN = 0;
    VIEW = 1;
    ADD = 2;
    BUY = 3;
  }

  ActionType action = 1;
  .google.protobuf.Timestamp time = 2;
  int64 item_id = 3;
  int64 user_id = 4;
  bytes extra_info = 5;
}
